File: src/unidade01/Exercicio01/algortimo02.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>

bool temColisao(double xa, double ya, double ra, double xb, double yb, double rb) {
    double distancia = sqrt(pow(xa - xb, 2) + pow(ya - yb, 2));
    return distancia <= (ra + rb);
}

int main() {
    if (temColisao(0, 0, 2, 3, 0, 2))
        printf("Colisão detectada.\n");
    else
        printf("Sem colisão.\n");

    if (temColisao(0, 0, 1, 5, 0, 1))
        printf("Colisão detectada.\n");
    else
        printf("Sem colisão.\n");
        
    return 0;
}

File: src/unidade01/Exercicio01/algortimo02.py
import math

def temColisao(xa, ya, ra, xb, yb, rb):
    distancia = math.sqrt((xa - xb)**2 + (ya - yb)**2)
    return distancia <= (ra + rb)

print(temColisao(0, 0, 2, 3, 0, 2))  # True, colisão
print(temColisao(0, 0, 1, 5, 0, 1))  # False, sem colisão
